slsim/Sources/source.py

```diff
@@ -140,6 +169,121 @@ def kwargs_variability_extracted(self):
+                            )
+
+                    # Create the agn object
+                    self.agn_class = agn.RandomAgn(
+                        self.agn_known_band,
+                        self.agn_known_mag,
+                        z,
+                        cosmo=self.cosmo,
+                        lightcurve_time=self.lightcurve_time,
+                        agn_driving_variability_model=self.agn_driving_variability_model,
+                        agn_driving_kwargs_variability=self.agn_driving_kwargs_variability,
+                        random_seed=random_seed,
+                        input_agn_bounds_dict=input_agn_bounds_dict,
+                        **agn_kwarg_dict
+                    )
+                    # Get mean mags for each provided band
+                    # determine which kwargs_variability are lsst bands
+                    lsst_bands = ["u", "g", "r", "i", "z", "y"]
+                    provided_lsst_bands = set(lsst_bands) & set(self.kwargs_variability)
+
+                    # The set "provided_lsst_bands" is no longer ordered.
+                    # Therefore, create a list of speclite names in the new order
+                    speclite_names = []
+
+                    # change name to be compatible with speclite filter names
+                    for band in provided_lsst_bands:
+                        speclite_names.append("lsst2023-" + band)
+
+                    # determine mean magnitudes for each band
+                    mean_magnitudes = self.agn_class.get_mean_mags(speclite_names)
+
+                    # Calculate light curve in each band
+                    for index, band in enumerate(provided_lsst_bands):
+
+                        # Define name for point source mags
+                        filter_name = "ps_mag_" + band
+
+                        # Set the filter to use
+                        self.agn_class.variable_disk.reprocessing_kwargs[
+                            "speclite_filter"
+                        ] = speclite_names[index]
+
+                        # Set the mean magnitude of this filter
+                        self.agn_class.variable_disk.driving_signal_kwargs[
+                            "mean_magnitude"
+                        ] = mean_magnitudes[index]
+
+                        # Extract the reprocessed light curve
+                        reprocessed_lightcurve = reprocess_with_lamppost_model(
+                            self.agn_class.variable_disk
+                        )
+
+                        # Prepare the light curve to be extracted
+                        times = reprocessed_lightcurve["MJD"]
+                        magnitudes = reprocessed_lightcurve[
+                            "ps_mag_" + speclite_names[index]
+                        ]
+
+                        # Prepare the time variable magnitude
+                        new_column = Column([min(magnitudes)], name=filter_name)
```

Comment:
agn class uses ps_mag_i from the source_dict. Then, it gives a lightcurve in i-band. Does peak magnitude of this lightcurve match the ps_mag_i that you used in agn class? ps_mag_i that is provided in source_dict is a peak magnitude of the quasar. If ps_mag_i and min(magnitudes) are suppose to be the same quantity, I think we should not replace ps_mag_i by min(magnitudes).